File: 3070-count-submatrices-with-top-left-element-and-sum-less-than-k/3070-count-submatrices-with-top-left-element-and-sum-less-than-k.js
/**
 * @param {number[][]} grid
 * @param {number} k
 * @return {number}
 */
var countSubmatrices = function(grid, k) {
    
    for( let i = 0 ; i < grid[0].length ; i++ ){
        for( let j = 1 ; j < grid.length ; j++ ){
            grid[j][i] += grid[j-1][i];
        }
    }

    let result = 0 ;

    for( let i = 0 ; i < grid.length ; i++ ){

        let sum = 0 ;
        let index = 0 ;

        while( index < grid[0].length ){

            sum += grid[i][index] ;
            if( sum > k ) break ;

            result ++ ;
            index ++ ;

        }
    }

    return result ;
};

File: 3070-count-submatrices-with-top-left-element-and-sum-less-than-k/3070-count-submatrices-with-top-left-element-and-sum-less-than-k.cpp
class Solution {
public:
    int countSubmatrices(vector<vector<int>>& grid, int k) {
        
        for( int i = 0 ; i < grid[0].size() ; i++ ){
            for( int j = 1 ; j < grid.size() ; j++ ){
                grid[j][i] += grid[j-1][i] ;
            }
        }

        int result = 0 ;

        for( int i = 0 ; i < grid.size() ; i++ ){

            int sum = 0 ;
            int index = 0 ;

            while( index < grid[0].size() ){

                sum += grid[i][index] ;
                if( sum > k ) break ;
                result ++ ;
                index ++ ;
            }
        }

        return result ;
    }
};